tests/protocols/test_websocket.py

```diff
@@ -658,3 +658,52 @@ async def send_text(url):
         await send_text("ws://127.0.0.1:8000")
 
     assert frames == [b"abc", b"abc", b"abc"]
+
+
+@pytest.mark.anyio
+@pytest.mark.parametrize("ws_protocol_cls", [WebSocketProtocol])
+@pytest.mark.parametrize("http_protocol_cls", HTTP_PROTOCOLS)
+async def test_server_can_read_messages_in_buffer_after_server_close(
+    ws_protocol_cls, http_protocol_cls
+):
+    """
+    Note: this doesn't work with WSProtocol.
```

Comment:
I'm confused. :sweat_smile: 

I've pulled this branch, and ran this test on both implementations to see why it was failing... But it actually passed with  `wsproto`.

Why did you add this note? :thinking: